acceptance/tests/rototillerTask/environment_variables.rb

```diff
@@ -9,23 +9,46 @@ def create_rakefile_task_segment(envs)
     segment = ''
     envs.each do |env|
       sut.add_env_var(env[:name], "#{env[:name]}: present value") if env[:exists]
-      segment += "t.add_env('#{env[:name]}'"
-      segment += ", '#{env[:default]}'" if env[:default]
-      segment += ", '#{env[:message]}'"
-      segment += '); '
-      segment += "puts ENV['#{env[:name]}']; "
+      if env[:block_syntax]
+        segment += "t.add_env do |env|\n"
+        remove_reserved_keys(env).each do |k, v|
+          segment += "  env.#{k.to_s} = '#{v}'\n"
+        end
+        segment += "end\n"
+      else
+        segment += "  t.add_env({"
+        remove_reserved_keys(env).each do |k, v|
+          segment += ":#{k} => '#{v}',"
+        end
+        segment += "})\n"
+      end
     end
     return segment
   end
 
+  def remove_reserved_keys(h)
+    hash = h.dup
+    [:block_syntax, :exists].each do |key|
+      hash.delete(key)
+    end
+    return hash
+  end
+
   step 'For environment variables that will succeed' do
     env_vars = [
       {:name => 'NO_DEFAULT-EXISTS',    :message => 'no default, previously exists',
-       :default => nil,                 :exists => true},
-       {:name => 'DEFAULT-EXISTS',       :message => 'default, previously exists',
-        :default => 'present default value',   :exists => true},
-        {:name => 'DEFAULT-NO_EXISTS',    :message => 'default, does notpreviously exist',
-         :default => 'DEFAULT-NO_EXISTS: notpresent default value', :exists => false},
+       :exists => true},
+      {:name => 'DEFAULT-EXISTS',       :message => 'default, previously exists',
+       :default => 'present default value',   :exists => true},
+      {:name => 'DEFAULT-NO_EXISTS',    :message => 'default, does notpreviously exist',
+       :default => 'DEFAULT-NO_EXISTS: notpresent default value', :exists => false},
+      {:name => 'NO_DEFAULT-EXISTS-BLOCK',    :message => 'no default, previously exists',
+       :exists => true, :block_syntax => true},
+      {:name => 'DEFAULT-EXISTS-BLOCK',       :message => 'default, previously exists',
+       :default => 'present default value',   :exists => true, :block_syntax => true},
+      {:name => 'DEFAULT-NO_EXISTS-BLOCK',    :message => 'default, does notpreviously exist',
+       :default => 'DEFAULT-NO_EXISTS-BLOCK: notpresent default value', :exists => false,
```

Comment:
some inconsistency in the formatting here.  i suspect it's largely because DEFAULT-EXISTS doesn't have its name in default.  but maybe that was by design, i don't recall